### DISCUSSION TITLE
HTODemux: Too many negatives after dehashing

### DISCUSSION THREAD

**Discussion by royfrancis** [satijalab/seurat#7628] [General]
I have a dataset with two hashed samples. Iget too many negatives after HTODemux dehashing with default settings.

```
      Type Cells
1  Doublet    30
2 Negative   764
3  Singlet   451
```

Nearly 61% are negatives.

```
            Doublet Negative Singlet
  TS-A-0301       8        8     275
  TS-A-0302      22      756     176
```

And nearly all the negatives are in TS-A-0302.

Based on violin plots, the negatives seem to have slightly lower mean UMIs than Singlets.

<img src="https://github.com/satijalab/seurat/assets/5083500/efa485c0-4338-4103-ae8c-77cfbae85504" width="350">

Here are the ridge plots and scatterplot of the two antibody levels.

<img src="https://github.com/satijalab/seurat/assets/5083500/80619060-de18-47d8-99cb-8268756eef83" width="220">

<img src="https://github.com/satijalab/seurat/assets/5083500/d08339aa-3f8b-4ce6-8b9a-da20a099e5cf" width="350">

Based on these two plots, I think the issue is that the overall antibody level of 0302 is much lower than 0301. 0302 levels are also not clearly separated from zero. And the automatically estimated threshold is probably too high for 0302.

Changing `kfunc` method in `HTODemux()` did not make a big difference. So, I am considering lowering `positive.quantile`. This does seem to increase Singlets and Doublets while decreasing Negatives.

Here are some comparisons:

`positive.quantile=0.99`

```
Cutoff for TS-A-0301 : 11 reads
Cutoff for TS-A-0302 : 1760 reads

            Doublet Negative Singlet
  TS-A-0301       8        8     275
  TS-A-0302      22      756     176
```

`positive.quantile=0.90`

```
Cutoff for TS-A-0301 : 7 reads
Cutoff for TS-A-0302 : 1170 reads
           
            Doublet Negative Singlet
  TS-A-0301      11        8     272
  TS-A-0302     114      454     386
```

`positive.quantile=0.80`

```
Cutoff for TS-A-0301 : 6 reads
Cutoff for TS-A-0302 : 965 reads
           
            Doublet Negative Singlet
  TS-A-0301      14        8     269
  TS-A-0302     177      340     437
```

`positive.quantile=0.70`

```
Cutoff for TS-A-0301 : 5 reads
Cutoff for TS-A-0302 : 833 reads

            Doublet Negative Singlet
  TS-A-0301      17        8     266
  TS-A-0302     272      249     433
```

`positive.quantile=0.60`

```
Cutoff for TS-A-0301 : 4 reads
Cutoff for TS-A-0302 : 730 reads
           
            Doublet Negative Singlet
  TS-A-0301      19        7     265
  TS-A-0302     421      164     369
```

It seems like lowering `positive.quantile` does increase singlets and doublets too? How low can I go? What are the considerations? Since the picture wasn't very clear, I decided to tune this parameter.

<img src="https://github.com/satijalab/seurat/assets/5083500/d82f18ba-e5b0-48d2-9a6e-5cca8a86838b" width="800">

X axis shows the threshold and Y axis shows the number of cells demultiplexed. The profiles for the two hashes are very different. 0302 looks quite wonky. Not sure if this is normal or something has gone wrong. 

Anyway, the default threshold is 0.99 (red vertical line) which gives

```
            Doublet Negative Singlet
  TS-A-0301       8        8     275
  TS-A-0302      22      756     176
```

I picked a new threshold 0.865 (vertical green line) which gives

```
            Doublet Negative Singlet
  TS-A-0301      13        8     270
  TS-A-0302     118      413     423
```

The threshold doesn't seem to have a huge effect on 0301, but it changes 0302 quite dramatically. 0.865 threshold maximises singlets while limiting the number of doublets.

This increases the total number of Singlet cells by 242 compared to default. What exactly is going on in this graph? Is this a reasonable thing to do? Can someone who understands the inner workings of `HTODemux()` please elaborate on this?

**Comment by artsvendsen** [comment]
Hi @royfrancis, it's been a minute since you last posted this and I'm facing a very similar problem with my dataset. Could you share some thoughts on how you proceed with your analysis? Thanks in advance!
